frontend/src/visualizations/index.js

```diff
@@ -0,0 +1,37 @@
+
+import Scalar     from "./Scalar.jsx";
+import Table      from "./Table.jsx";
+import LineChart  from "./LineChart.jsx";
+import BarChart   from "./BarChart.jsx";
+import PieChart   from "./PieChart.jsx";
+import AreaChart  from "./AreaChart.jsx";
+import USStateMap from "./USStateMap.jsx";
+import WorldMap   from "./WorldMap.jsx";
+import PinMap     from "./PinMap.jsx";
+
+const visualizations = new Map();
+
+export function registerVisualization(visualization) {
+    let identifier = visualization.identifier;
+    if (identifier == null) {
+        throw new Error("Visualization must define an 'identifier' static variable: " + visualization.name);
+    }
+    if (visualizations.has(identifier)) {
+        throw new Error("Visualization with that identifier is already registered: " + visualization.name);
+    }
+    visualizations.set(identifier, visualization);
+}
+
+registerVisualization(Scalar);
+registerVisualization(Table);
+registerVisualization(LineChart);
+registerVisualization(BarChart);
+registerVisualization(PieChart);
+registerVisualization(AreaChart);
+registerVisualization(USStateMap);
+registerVisualization(WorldMap);
+registerVisualization(PinMap);
+
+window.visualizations = visualizations;
```

Comment:
why is it necessary to set this on window?  if we are doing this we probably should prefix it with something more specific like `MetabaseVisualizations` to make sure it won't clash with anything else.